src/mock_etcd.rs

```diff
@@ -128,22 +155,129 @@ impl MockEtcd {
         kvs
     }
 
+    /// Send watch response for a specific watch id to etcd client
+    async fn send_watch_response_with_watch_id(
+        watch_response_sender: Arc<RwLock<DuplexSink<WatchResponse>>>,
+        kv: KeyValue,
+        prev_kv: Option<KeyValue>,
+        watch_id: i64,
+        event_type: Event_EventType,
+    ) {
+        let mut event = Event::new();
+        event.set_field_type(event_type);
+        if let Some(value) = prev_kv {
+            event.set_prev_kv(value);
+        }
+        event.set_kv(kv);
+
+        let mut response = WatchResponse::new();
+        response.set_watch_id(watch_id);
+        response.set_events(RepeatedField::from_vec([event].to_vec()));
+
+        watch_response_sender
+            .write()
+            .await
+            .send((response, WriteFlags::default()))
+            .await
+            .unwrap_or_else(|e| panic!("Fail to send watch response, the error is {}", e));
+    }
+
+    /// Send watch response to etcd client
+    #[allow(clippy::pattern_type_mismatch)]
+    #[allow(clippy::type_complexity)]
+    async fn send_watch_responses(
+        watch_arc: Arc<RwLock<HashMap<i64, KeyRange>>>,
+        watch_response_sender_arc: Arc<
+            RwLock<HashMap<i64, Arc<RwLock<DuplexSink<WatchResponse>>>>>,
+        >,
+        kv: KeyValue,
+        prev_kv: Option<KeyValue>,
+        event_type: Event_EventType,
+    ) {
+        // Find all watch ids which watch this key and send watch response
+        let watch = watch_arc.read().await;
+        let watch_response_sender = watch_response_sender_arc.read().await;
+        for (watch_id, v) in watch.iter() {
+            let sender = watch_response_sender
```

Comment:
This get operation is not efficient for a large watch list for that only a small part or even none of them needs to send a watch response.

Reply:
Yes, it is just for test and a very easy solution. If needed, we can implement it by range tree in future.